analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Binary.cs

```diff
@@ -1082,8 +1082,9 @@ public void Binary_BitXor_SingleValue(int left, int right, int expected)
     [DataRow("i >=  4 && i <=  6 && j >=  6", 0, null)]
     [DataRow("i >=  4 && i <=  6 && j >=  6 && j <= 8", 0, 15)]   // exact range: 0, 14
     [DataRow("i >=  4 && i <=  5 && j >=  6 && j <= 8", 1, 15)]   // exact range: 2, 13
-    [DataRow("i >= -3 && i <= -1 && j >= -4 && j <=-2", 0, null)] // exact range: 0, 3
-    [DataRow("i >= -4 && i <= -3 && j >= -2 && j <=-1", 1, null)] // exact range: 2, 3
+    [DataRow("i >= -3 && i <= -1 && j >= -4 && j <=-2", 0, 7)]    // exact range: 0, 3
+    [DataRow("i >= -4 && i <= -3 && j >= -2 && j <=-1", 1, 7)]    // exact range: 2, 3
+    [DataRow("i >= -4 && i <=  6 && j >= -3 && j <= 8", -8, 15)]
```

Comment:
The exact range for the new interval is $`[-12, 14]`$.
The range found is $`[-8, 15]`$, which doesn't contain the exact range on the negative side.
If we take -12, for example: $`-4 \oplus 8 = -12 \notin [-8, 15]`$

Reply:
Very good catch. Nasty edge cases. I added a comment for further explanations.

Reply:
Very good!